File: jnet-biz/src/main/java/com/jnet/biz/entity/Project.java
package com.jnet.biz.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.io.Serial;
import java.io.Serializable;
import java.time.LocalDateTime;

/**
 * 项目管理实体
 *
 * @author JNet Team
 * @since 2024-04-16
 */
@Data
@EqualsAndHashCode(callSuper = false)
@TableName("biz_project")
public class Project implements Serializable {

    @Serial
    private static final long serialVersionUID = 1L;

    /**
     * 主键ID
     */
    @TableId(value = "project_id", type = IdType.AUTO)
    private Long projectId;

    /**
     * 项目名称
     */
    private String name;

    /**
     * 项目编码（唯一）
     */
    private String code;

    /**
     * 负责人ID
     */
    private Long managerId;

    /**
     * 伦理批件号
     */
    private String ethicsCode;

    /**
     * 隐私级别 (1:公开, 2:脱敏, 3:绝密)
     */
    private Integer privacyLevel;

    /**
     * 项目描述
     */
    private String description;

    /**
     * 目标检测类别配置 (JSONB)
     */
    private String targetClasses;

    /**
     * 状态 (active/archived/deleted)
     */
    private String status;

    /**
     * 创建人ID
     */
    private Long createBy;

    /**
     * 创建时间
     */
    private LocalDateTime createTime;

    /**
     * 更新人ID
     */
    private Long updateBy;

    /**
     * 更新时间
     */
    private LocalDateTime updateTime;

    /**
     * 删除标志
     */
    @TableLogic
    private Boolean delFlag;
}

File: ui/src/types/project.ts
/**
 * 项目相关类型定义
 */

// 项目实体
export interface Project {
  projectId: number
  name: string
  code: string
  managerId?: number
  ethicsCode?: string
  privacyLevel?: number  // 1:公开, 2:脱敏, 3:绝密
  description?: string
  targetClasses?: string // JSON字符串
  status?: string  // active/archived/deleted
  createBy?: number
  createTime: string
  updateBy?: number
  updateTime: string
  delFlag?: boolean
}

// 项目VO（视图对象）
export interface ProjectVO extends Project {
  // 可以扩展额外的展示字段
  stats?: {
    batchCount: number
    imageCount: number
    annotatedCount: number
  }
}

// 项目DTO（创建/更新）
export interface ProjectDTO {
  name: string
  code: string
  managerId?: number
  ethicsCode?: string
  privacyLevel?: number  // 1:公开, 2:脱敏, 3:绝密
  description?: string
  targetClasses?: string
  status?: string  // active/archived/deleted
}

// 项目查询DTO
export interface ProjectQueryDTO {
  current: number
  size: number
  name?: string
  status?: string  // active/archived/deleted
  managerId?: number
  startDate?: string
  endDate?: string
}
